docs/project_setup_and_build.md

```diff
@@ -0,0 +1,229 @@
+# Project structure and build
+
+## Overview
+
+In this chapter, you will get to know the anatomy of the standard BigFlow project, deployment artifacts, and [CLI commands](./cli.md).
+
+The BigFlow build command packages your processing code into a Docker image. Thanks to this approach, you can create any
+environment you want for your pipelines. What is more, you don't need to worry about dependencies clashes on the Cloud Composer (Airflow).
+
+The scheme of a BigFlow deployment artifacts looks like this:
+
+![BigFlow artifact scheme](./images/bigflow-artifact.png)
+
+Your project is turned into a standard Python package (which can be uploaded to [pypi](https://pypi.org/) or installed locally using `pip`). 
+Next, the package is closed into a Docker image with fixed Python version. Finally, there are Airflow DAGs that uses this image.
+
+A single DAG is generated based on the `bigflow.workflow.Workflow` class. 
+Produced DAG consists only of [`KubernetesPodOperator`](https://airflow.apache.org/docs/stable/_api/airflow/contrib/operators/kubernetes_pod_operator/index.html) objects, which
+execute operations on a Docker image.
+
+To build a project you need to use the `bigflow build` command. The documentation you are reading is also a valid BigFlow
+project. Go to the `docs` directory and run the `bigflow build` command to see how the build process works. The `bigflow build`
+command should produce three artifacts:
+
+* The `dist` directory with a Python package
+* The `image` directory with a deployment configuration and, optionally, with a Docker image as `.tar`
+* The `build` directory with JUnit test results
+
+The `bigflow build` command uses three subcommands to generate all the artifacts: `bigflow build-package`, `bigflow build-image`, `bigflow build-dags`.
+
+Now, let us go through each building element in detail, starting from the Python package.
+
+## Package
+
+### Project structure
+
+A example BigFlow project, with the standard structure, looks like this:
+
+```
+project_dir/
+    project_package/
+        __init__.py
+        workflow.py
+    test/
+        __init__.py
+    resources/
+        requirements.txt
+    Dockerfile
+    project_setup.py
+    deployment_config.py
+```
+
+Let us start with `project_package`. It's a Python package that contains the processing logic of your pipelines.
+It also contains `Workflow` objects, that arranges parts of your processing logic into a [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph).
+Read the [Workflow & Job chapter](./workflow-and-job.md) to learn more about workflows and jobs.
+The `project_package` is used to create a standard Python package that can be installed using `pip`.
+
+The `project_setup.py` is the build script for the project. It turns the `project_package` into a `.whl` package. 
```

Comment:
moim zdaniem można napisać
**The `project_setup.py` file** lub **`project_setup.py`** jeśli traktujesz to jak nazwę własną ale nie **The `project_setup.py`**

musisz zdecydować czy traktujesz to jak rzeczownik (file) czy jak nazwę własną

Reply:
done (nazwa własna)